### PR TITLE
Refactored Vulnerability Builder

### LINKED ISSUE #2562: Refactor vulnerability_builder and kics/service
### Description
Refactor vulnerability_builder  and kics/service for easier code read

### PR DESCRIPTION
**Pull request by joaoReigota1**
Added Detector package

Closes #2562

**Proposed Changes**

- Refactored Vulnerability Builder
- Added Detector package
- Increased kics/service.go coverage

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

**Comment by rogeriopeixotocx** [comment]
Closes #2435

File: pkg/model/model.go [hunk 1]
```diff
@@ -49,6 +49,13 @@ var (
 	}
 )
 
+// VulnerabilityLines is the representation of the found line for issue
+type VulnerabilityLines struct {
+	Line                 int
+	VulnLine             []VulnLines
```
Comment (rogeriopeixotocx) [inline pkg/model/model.go:55]:
```suggestion
	VulnLines             []VulnLine
```

Comment (rogeriopeixotocx) [reply pkg/model/model.go:55]: I wonder if we should rename this internal struct to `[]CodeLine`

WDYT @felipe-avelar @joaoReigota1?

File: pkg/model/model.go [hunk 2]
```diff
@@ -91,27 +98,27 @@ type QueryMetadata struct {
...
+	FileName         string      `db:"file_name" json:"fileName"`
+	QueryID          string      `db:"query_id" json:"queryID"`
+	QueryName        string      `db:"query_name" json:"queryName"`
+	QueryURI         string      `json:"-"`
+	Category         string      `json:"category"`
+	Description      string      `json:"description"`
+	Platform         string      `db:"platform" json:"platform"`
+	Severity         Severity    `json:"severity"`
+	Line             int         `json:"line"`
+	VulnLines        []VulnLines `json:"vulnLines"`
```
Comment (rogeriopeixotocx) [inline pkg/model/model.go:114]: Naming concerns addressed in previous comments

File: pkg/detector/helper.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+			Position: startPos,
+		}
+		startPos++
+	}
+	return vulns
+}
+
+// SelectLineWithMinimumDistance will search a map of levenshtein distances to find the minimum distance
+func SelectLineWithMinimumDistance(distances map[int]int, startingFrom int) int {
+	minDistance, lineOfMinDistance := 1000000000000, startingFrom
```
Comment (rogeriopeixotocx) [inline pkg/detector/helper.go:156]: Do we have a constant for integer's max number in golang?

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
This looks good, just a few questions regard naming and constants.

File: pkg/model/model.go (hunk 1 above)
Comment (joaoReigota1) [reply pkg/model/model.go:55]: Agree, will change to []CodeLine

**Review by felipe-avelar — COMMENTED** [review]
I really liked the approach used on `sink` and `resolveSink`. 😃 

Besides @rogeriopeixotocx comments, this PR looks good for me.

File: pkg/detector/helper.go (hunk 3 above)
Comment (rogeriopeixotocx) [reply pkg/detector/helper.go]: https://golang.org/pkg/math/#pkg-constants

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**